modules/machineset-creating.adoc

```diff
@@ -0,0 +1,261 @@
+        sigs.k8s.io/cluster-api-machineset: <cluster_name>-<machine_label>-<AWS-availability-zone> <1>
+...
+----
+<1> Ensure that both the `sigs.k8s.io/cluster-api-machineset` parameter values
+match the `name` that you defined in the `metadata` section.
+
+. In `<file_name>.yaml`, add the node label definition to the spec. The label
+definition resembles the following stanza:
++
+[source,yaml]
+----
+  spec:
+    metadata:
+      labels:
+        node-role.kubernetes.io/<label_name>: "" <1>
+----
+<1> In this definition, `<label_name>` is the node label to add. For example, to
+add the `infra` label to the nodes, specify `node-role.kubernetes.io/infra`.
++
+The updated `spec` section resembles this example:
++
+[source,yaml]
+----
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      sigs.k8s.io/cluster-api-cluster: cluster_name
+      sigs.k8s.io/cluster-api-machineset: <cluster_name>-<machine_label>-<AWS-availability-zone>
+  template:
+    metadata:
+      creationTimestamp: null
+      labels:
+        sigs.k8s.io/cluster-api-cluster: <cluster_name>
+        sigs.k8s.io/cluster-api-machine-role: <machine_label>
+        sigs.k8s.io/cluster-api-machine-type: <machine_label>
+        sigs.k8s.io/cluster-api-machineset: <cluster_name>-<machine_label>-<AWS-availability-zone>
+    spec: <1>
+      metadata:
+        labels:
+          node-role.kubernetes.io/<label_name>: ""
+...
+----
+<1> Place the `spec` stanza here.
+
+. Optionally, modify the EC2 instance type and modify the storage volumes.
++
+[IMPORTANT]
+====
+Take care to modify only the parameters that describe the EC2 instance type
+and storage volumes. You must not change the other parameters value in the
+`providerSpec` section.
+====
++
+[source,yaml]
+----
+providerSpec:
+  value:
+    ami:
+      id: ami-0e2bcd33dfff9c73e <1>
+    apiVersion: awsproviderconfig.k8s.io/v1alpha1
+    blockDevices: <2>
+    - ebs:
+        iops: 0
+        volumeSize: 120
+        volumeType: gp2
+    deviceIndex: 0
+    iamInstanceProfile: <3>
+      id: <cluster_name>-<machine_label>-profile
+    instanceType: m4.large <4>
+    kind: AWSMachineProviderConfig
+    metadata:
+      creationTimestamp: null
+    placement: <3>
+      availabilityZone: <AWS-availability-zone>
+      region: <AWS-region>
+    publicIp: null
+    securityGroups:
+    - filters:
+      - name: tag:Name
+        values:
+        - testcluster2_worker_sg
+    subnet: <3>
+      filters:
+      - name: tag:Name
+        values:
+        - <cluster_name>-<machine_label>-<AWS-availability-zone>
+    tags:
+    - name: openshiftClusterID
+      value: 5a21bfc0-1c56-4400-81bb-7fd66644f871
+    - name: kubernetes.io/cluster/<cluster_name>
+      value: owned
+    userDataSecret: <3>
+      name: <machine_label>-user-data
+----
+<1> You can specify a different valid AMI.
+<2> You can customize the volume characteristics for the MachineSet. See the AWS
+documentation.
+<3> Do not modify this section.
```

Comment:
Should you create new resources for the items in these sections (userDataSecret, subnet, placement, and iamInstanceProfile) to reference the new MachineSet?